### PR TITLE
feat: add todo test and todo logic

### PR DESCRIPTION
**Pull request by Dlutermade**
先添加好 Todo Test 和 Todo Logic

目前可能需要的 Logic
1. get emit event player hands (取得發出事件玩家手牌) 
1.1. 用於判斷使用者 是否必須 棄牌
1.2. 用於判斷使用者 棄掉的牌 是否出於 玩家手中

### CODE REVIEW

File: apps/backend/src/action/logic/pass.ts [hunk 1]
```diff
@@ -1,25 +1,55 @@
...
+   * 1. check player have cards left
+   *  if player not cards left and discard card eq null
+   *    - can Pass
+   *
+   * 2. check player have the card
+   *  get player hands if hands can find the discard card
+   *    - can Pass
+   *  else not find
+   *    - Error
+   **/
```
Comment (kayac-chang) [inline apps/backend/src/action/logic/pass.ts:52]: God Job!!!
Try to use English, and don't be afraid to make mistakes.

You can also consider using ChatGPT or Grammarly to check your sentence.
This definitely helps.

**Review by kayac-chang — CHANGES_REQUESTED** [review]

File: apps/backend/src/action/command/pass.command.ts [hunk 2]
```diff
@@ -4,13 +4,13 @@ import { PathCard } from "~/models/card";
...
   data: z.object({
-    card: z.nativeEnum(PathCard),
+    card: z.nativeEnum(PathCard).nullable(),
   }),
 });
 
 export type PassCommand = Readonly<z.infer<typeof PassCommandSchema>>;
 
-export function PassCommand(card: PathCard): PassCommand {
+export function PassCommand(card: PathCard | null): PassCommand {
```
[the remark this answers is not in the document]
Comment (Dlutermade) [reply apps/backend/src/action/command/pass.command.ts:13]: 我當時也在想要用 null 還是 ? 還是undefined

File: apps/backend/src/action/logic/pass.test.ts [hunk 3]
```diff
@@ -1,3 +1,119 @@
...
+    const read = jest.fn().mockImplementation(() => {
+      return Promise.resolve(store);
+    });
+
+    source = { append, read, on: jest.fn(), off: jest.fn() };
+  });
+
+  /**
+   * 由於目前抽牌事件是否要綁在這個流程還不確定
+   * 因此到時候可能移除
```
[the remark this answers is not in the document]
Comment (Dlutermade) [reply apps/backend/src/action/logic/pass.test.ts:22]: LAMO，I forgot to remove

File: apps/backend/src/action/command/pass.command.ts [hunk 4]
```diff
@@ -4,13 +4,13 @@ import { PathCard } from "~/models/card";
 export const PassCommandSchema = z.object({
   type: z.literal("pass"),
   data: z.object({
-    card: z.nativeEnum(PathCard),
+    card: z.nativeEnum(PathCard).nullable(),
```
[the remark this answers is not in the document]
Comment (Dlutermade) [reply apps/backend/src/action/command/pass.command.ts:7]: if use  `nullish` 
should we update  `card: PathCard | null` to  `card?: PathCard | null`

File: apps/backend/src/action/command/pass.command.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kayac-chang) [reply apps/backend/src/action/command/pass.command.ts]: LGTM

File: apps/backend/src/action/command/pass.command.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (kayac-chang) [reply apps/backend/src/action/command/pass.command.ts]: By using `nullish` which can handle both scenario `null` or `undefined`.

[the remark this answers is not in the document]
Comment (kayac-chang) [reply apps/backend/src/action/command/pass.command.ts]: LGTM

File: apps/backend/src/action/event/turn-has-been-passed.event.ts [hunk 5]
```diff
@@ -18,6 +18,8 @@ export function isTurnHasBeenPassedEvent(
   return TurnHasBeenPassedEventSchema.safeParse(event).success;
 }
 
-export function TurnHasBeenPassedEvent(card: PathCard): TurnHasBeenPassedEvent {
-  return { type: "turn has been passed", data: { card } };
+export function TurnHasBeenPassedEvent(
+  discardCard: PathCard | null
```
[the remark this answers is not in the document]
Comment (kayac-chang) [reply apps/backend/src/action/event/turn-has-been-passed.event.ts]: LGTM

File: apps/backend/src/action/event/turn-has-been-passed.event.ts [hunk 6]
```diff
@@ -4,7 +4,7 @@ import { PathCard } from "~/models/card";
 export const TurnHasBeenPassedEventSchema = z.object({
   type: z.literal("turn has been passed"),
   data: z.object({
-    card: z.nativeEnum(PathCard),
+    discardCard: z.nativeEnum(PathCard).nullable(),
```
[the remark this answers is not in the document]
Comment (kayac-chang) [reply apps/backend/src/action/event/turn-has-been-passed.event.ts]: LGTM

File: apps/backend/src/action/logic/pass.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (kayac-chang) [reply apps/backend/src/action/logic/pass.test.ts:22]: LGTM

**Review by kayac-chang — APPROVED** [review]
LGTM

**Pull request merged**